src/lib.rs

```diff
@@ -34,6 +34,7 @@ use pyo3::{
 use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 #[cfg(feature = "pyo3")]
 use std::collections::hash_map::DefaultHasher;
+use std::collections::HashSet;
```

Comment:
In general I am a big `FxHashSet` fan :)

Reply:
i'll use it once pep508_rs becomes performance critical